### PR TITLE
Remove Job from `docker kill`

### PR DESCRIPTION
**Pull request by duglin**
Signed-off-by: Doug Davis dug@us.ibm.com

### CODE REVIEW

File: api/server/server.go [hunk 1]
```diff
@@ -196,13 +196,14 @@ func postContainersKill(eng *engine.Engine, version version.Version, w http.Resp
...
 		return err
 	}
-	job := eng.Job("kill", vars["name"])
-	if sig := r.Form.Get("signal"); sig != "" {
-		job.Args = append(job.Args, sig)
-	}
-	if err := job.Run(); err != nil {
+
+	name := vars["name"]
+	sig := vars["signal"]
```
Comment (crosbymichael) [inline api/server/server.go]: I think the signal should be converted to an int in the API layer so that a StatusBadRequest can be returned for things like this as `int` is the final type for this var.

Comment (LK4D4) [inline api/server/server.go]: +1
Then we can remove `daemon.ContainerKill`

Comment (duglin) [inline api/server/server.go]: I thought about that, but I wasn't sure if we wanted callers to ContainerKill be able to pass in both ints and the string version w/o forcing everyone to convert it themselves.

Comment (LK4D4) [inline api/server/server.go]: I think we should enforce type. That was whole idea of removing jobs.

File: api/server/server.go [hunk 2]
```diff
@@ -193,16 +194,34 @@ func postContainersKill(eng *engine.Engine, version version.Version, w http.Resp
...
 	}
-	job := eng.Job("kill", vars["name"])
-	if sig := r.Form.Get("signal"); sig != "" {
-		job.Args = append(job.Args, sig)
+
+	var sig uint64
+	name := vars["name"]
+
+	// If we have a signal, look at it. Otherwise, do nothing
+	if sigStr := vars["signal"]; sigStr != "" {
```
Comment (LK4D4) [inline api/server/server.go]: What will be if sig == ""?
Can we move whole `ContainerKill` here to work with container instance?

Comment (duglin) [inline api/server/server.go]: If sig == "" then the normal container.Kill() will be done.

I thought about removing ContainerKill() but then each potential caller would need to do the logging themselves and I didn't like the idea of that.

File: daemon/kill.go [hunk 3]
```diff
@@ -55,7 +27,7 @@ func (daemon *Daemon) ContainerKill(job *engine.Job) error {
 		if err := container.KillSig(int(sig)); err != nil {
 			return fmt.Errorf("Cannot kill container %s: %s", name, err)
 		}
-		// FIXME: Add event for signals
+		container.LogEvent("kill signal:" + strconv.Itoa(int(sig)))
```
Comment (LK4D4) [inline daemon/kill.go]: I'm not sure that I want introduce new event in this PR.

File: api/server/server.go (hunk 2 above)
Comment (LK4D4) [inline api/server/server.go]: Ok, just looks weird comparing to other merged PRs. pause/unpause in particular. Also I think there won't be other callers.

File: daemon/kill.go (hunk 3 above)
Comment (duglin) [inline daemon/kill.go]: oh, is there a formal process to get events approved?  I assumed this was an old comment and would be easy to fix.  I'll revert it.

Comment (duglin) [inline daemon/kill.go]: actually, can I just reuse the same event from above?

File: api/server/server.go (hunk 2 above)
Comment (duglin) [inline api/server/server.go]: This might be a bigger discussion (bordering on bikeshedding :-) ) but to me the api/server code is all about the http transport and I expect we'll eventually have others, so when that day comes I would want logic that isn't specific to the transport to be done within the daemon itself.  And I consider logging of a container being deleted to be the job of the deamon (or the container itself) but not the api/http/transport layer.  So, in that respect I would prefer if we didn't remove those other daemon funcs, if nothing else to have consistency and proper separation of concern.

File: daemon/kill.go (hunk 3 above)
Comment (LK4D4) [inline daemon/kill.go]: I think it should be sort of separate PR.
I don't know how to post signal to events - it is sorta break events format.

**Comment by duglin** [comment]
ok I think I addressed all of the comments.

**Comment by cpuguy83** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM
